pkg/k8s/watchers/pod.go

```diff
@@ -719,6 +719,11 @@ func (k *K8sWatcher) deleteHostPortMapping(pod *slim_corev1.Pod, podIPs []string
 	}
 
 	for _, dpSvc := range svcs {
+		svcs := k.svcManager.GetDeepCopyServices()
```

Comment:
The ownership check also warrant a code comment explaining the motivation.

Reply:
Done!